app/api/log/route.ts

```
@@ -0,0 +1,42 @@
+    }
+
+    try {
+        const body = await req.json();
+        const { message, data } = body;
+
+        if (!message || typeof message !== 'string') {
+            return NextResponse.json({ success: false, error: 'Invalid message format' }, { status: 400 });
+        }
+
+        // Sanitize message: limit length and strip control characters
+        const sanitizedMessage = message.slice(0, 1000).replace(/[\x00-\x1F\x7F]/g, '');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)